addon/-private/utils/parse-response-headers.ts

```diff
@@ -1,7 +1,11 @@
 const newline = /\r?\n/;
 
-export default function parseResponseHeaders(headersString) {
-  let headers = Object.create(null);
+interface ResponseHeaders {
+  [key: string]: string
+};
```

Comment:
This type is way more foundational than just "response headers". 

Let's create a `..../-private/types.ts` and have a

```ts
export interface Dict<T = unknown> {
   [k: string]: T | undefined;
}
```

and then you can import it and

```ts
export default function parseResponseHeaders(headersString: Dict<string>)
```

Reply:
would that not be something you'd want in the global index.d.ts?

Reply:
Nope. Consumers of the ember-data won't get any of those types.

Reply:
I'm not sure you understand. Everything in your `/types` folder is for _your project only_. If you put `Dict` there, and used it anywhere in parts of the library that users of ember-data touch, it'll cause them problems.

Reply:
right, in my latest commit, I moved the Dict type to -private/types.ts, and it didn't need to be imported. :-\